### ISSUE TITLE
magento/framework/App/Language/Dictionary.php fails to skip empty csv line

### ISSUE THREAD

**Issue by romeof1980** [magento/magento2#25609]
### Preconditions (*)
1. Magento 2.2.2
2. Php 7.1

### Steps to reproduce (*)
1. set up a csv translation file with some translations and an empty row 
2. load a page where it is requested (frontend / backend)

### Expected result (*)
1. Page loads

### Actual result (*)
1. Exception is thrown
Undefined offset: 1 in vendor/magento/framework/App/Language/Dictionary.php on line 196

the reason is that in the following code (lines 195 - 197) the $row[1] is null
```
while (($row = $file->readCsv()) !== false) {
                    $result[$row[0]] = $row[1];
                }
```

should be checked if $row[1] is null before assigning the value to $result[$row[0]]

**Comment by helenapaes** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @helenapaes. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @helenapaes, here is your [Magento instance](https://i-25609-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-25609-2-3-develop.instances.magento-community.engineering/admin](https://i-25609-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by engcom-Echo** [comment]
Hello @romeof1980 

Thank you for contribution and collaboration!

We are not able to reproduce this issue on the 2.3-develop branch by provided steps.

Testing scenario:

Steps to reproduce:
       

1. Create simple module with controller which consists translated phrases
2. Set up a csv translation file with some translations and an empty row
![Screenshot from 2019-11-15 19-10-46](https://user-images.githubusercontent.com/51681435/68961790-ce624380-07db-11ea-8ec4-b9e779265f24.png)
3. Load a page where it is requested (frontend / backend)

Result: 
Page is loaded successfully 
![Screenshot from 2019-11-15 19-17-59](https://user-images.githubusercontent.com/51681435/68962220-bb03a800-07dc-11ea-8cb9-738565bd2336.png)

Thank you!

**Comment by engcom-Echo** [comment]
Hello @romeof1980 

Thank you for youre report!

Fix for this issue is available on 2.2.9 and 2.3.3 release, please try upgrade.
lib/internal/Magento/Framework/App/Language/Dictionary.php
```
     while (($row = $file->readCsv()) !== false) {
                    if (is_array($row) && count($row) > 1) {
                        $result[$row[0]] = $row[1];
                    }
                }
```
So I have to close this issue.
Thank you!
